skbeam/core/roi.py

```diff
@@ -398,50 +398,55 @@ def mean_intensity(images, labeled_array, index=None):
 
 
-def circular_average(image, calibrated_center, threshold=0, nx=100,
-                     pixel_size=(1, 1),  min_x=None, max_x=None):
+def circular_average(image, r_array, pixel_size=(1, 1), statistic='mean',
+                     min_x=None, max_x=None, mask=None, bins=None):
     """Circular average of the the image data
-    The circular average is also known as the radial integration
+    The circular average is also known as the radial or azimuthal integration
     Parameters
     ----------
     image : array
         Image to compute the average as a function of radius
-    calibrated_center : tuple
-        The center of the image in pixel units
-        argument order should be (row, col)
-    threshold : int, optional
-        Ignore counts above `threshold`
-        default is zero
-    nx : int, optional
-        number of bins in x
-        defaults is 100 bins
+    r_array : array
+        Array which relates pixel positions to radial positions
     pixel_size : tuple, optional
         The size of a pixel (in a real unit, like mm).
         argument order should be (pixel_height, pixel_width)
         default is (1, 1)
+    statistic: str or func, optional
+        The statistic to compute over the integration, defaults to mean
     min_x : float, optional number of pixels
         Left edge of first bin defaults to minimum value of x
     max_x : float, optional number of pixels
         Right edge of last bin defaults to maximum value of x
+    mask: bool array, optional
+        The array of pixels to be removed from the image before integration
+    bins: array, optional
+        The bins to use in the integration, if none given the function will
+        give its best assessment based on the pixel_size and r_array
     Returns
     -------
     bin_centers : array
-        The center of each bin in R. shape is (nx, )
-    ring_averages : array
-        Radial average of the image. shape is (nx, ).
+        The center of each bin in R
+    int_stat : array
+        Radial integrated statistic of the image.
     """
-    radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size)
-
-    bin_edges, sums, counts = utils.bin_1D(np.ravel(radial_val),
-                                           np.ravel(image), nx,
-                                           min_x=min_x,
-                                           max_x=max_x)
-    th_mask = counts > threshold
-    ring_averages = sums[th_mask] / counts[th_mask]
-
-    bin_centers = utils.bin_edges_to_centers(bin_edges)[th_mask]
-
-    return bin_centers, ring_averages
+    # If there is no mask, make one
+    if mask is None:
+        mask = np.ones(image.shape, dtype=int).astype(bool)
+
+    if bins is None:
+        res = np.sqrt(np.sum([a ** 2 for a in pixel_size]))
+        bins = np.arange(np.min(r_array) - res/2., np.max(r_array) + res/2.,
+                         res)
+    bin_edges, int_stat, bin_num = sts.binned_statistic(r_array[mask],
```

Comment:
Ah, yes I did forget that import, I think having scipy as a dep is fine, I mean it is part of the scientific python stack and it might even be a dep that is included in the conda recipe already.

I would prefer to not have to rewrite functions that already exist unless we are expecting that we can do it much better (faster, lower memory, etc.). Although I do know that a class version of binned_statistic is in the pipeline. We could expose a hook for this in the signature if we want.